README.md

```diff
@@ -88,3 +88,15 @@ For such binaries where test cases are independent, `gtest-parallel` provides
 `--serialize_test_cases` that runs tests within the same test case sequentially.
 While generally not providing as much speedup as fully parallel test execution,
 this permits such binaries to partially benefit from parallel execution.
+
+## XML Output
+
+This script is able to save test results in XML format. Use
```

Comment:
What XML format? Where does it come from? What's the specification? If this is a specific JUnit format, then please say so and link to the spec. Additionally I think we should call it something like `--dump_junit_test_results` instead of calling it generic "XML".

Also, what do you think about having something like `--output_format=junit` and `--output=results.xml` ? That way we could add new formats without adding new flags. Do you see a need for having multiple concurrent sinks?

Reply:
Well, the format resembles the one described here: https://github.com/google/googletest/blob/master/googletest/docs/AdvancedGuide.md
Intention to be parseable by JUnit plugin in Jenkins. I can refer to that ocument with outlining what was added.
If same switch is used for multiple formats, can it follow same approach as gtest and specify format right in the command, for example `--dump_test_results=junit:filename` with `--dump_junit_test_results` preserved for backward compatibility? I think I could have, say, `xml` and `xml-full` formats instead of separate format specific flag `--dump_xml_all_logs`.
I also think that only one test results sink is needed. In my code concurrent sink appeared from independent --dump_ flags.

Reply:
Looks like there already is precedence for an existing flag, per that page: "To generate the XML report, set the GTEST_OUTPUT environment variable or the --gtest_output flag to the string "xml:_path_to_output_file_", which will create the file at the given location."

If you're reusing a gtest flag with the same semantics as that gtest flag I think that lessens the burden of us to document the format / behavior (because it does what it's expected to do). We should still link to gtest documentation appropriately here.

So implementing `--gtest_output=xml:path_to_file.xml` makes sense to me here. There's no need to implement any other flags, skip `dump_xml_all_logs` until there's actually a need for it (and we figure out how we want it implemented). I think you can filter an xml file to drop passed results fairly cheaply and the cost of creating and storing it needs to be substantial for us to consider the all-logs argument (it should do the same as `--gtest_output` on the gtest binary does by default).

I'm not sure the existing JSON format is the same as the gtest json format. Unless you know this format is the same that gtest uses are the same, do not implement `--gtest_output=json:`. If they are the same then having it use gtest_output is better.

Reply:
I didn't want to use `--gtest_output` flag because generated XML is not exactly the same as the one produced by gtest. The difference is mainly in failures messages and captured standard output.
Let's try the following: use `--gtest_output` for xml output, generate xml file for each individual test, parse them and aggregate results into single xml output as user requested. I agree, it would be easier to document then. It will take some time for me to implement though as I'm doing this on my spare time.
I still need to add standard output to generated XML tests, I'm not happy with what I'm getting from gtest by default. Perhaps under `xml-full` format.
I'm not going to implement JSON format support. In fact, I didn't know it's supported now in gtest.

Reply:
Are the formats compatible (are you using the same tags, headers etc.) so that a parser for --gtest_output=xml:foo would read something reasonable from our output?

If so I think you can add --gtest_output and have it generate what you want instead of merging --gtest_output from the binaries. Is there a reasonable heading in the existing format where you can put stdout?

Reply:
How do the failure messages and and stdout reported in the formats differ?

Reply:
In gtest failure message contains same text as printed in gtest. In my XML it's just a message that test exited with non zero status. I don't like this, I'd rather have original gtest failure mesage. Stdout is not reported in gtest but if added as <system-out> subtag of testcase then Jenkins JUnit parser will understand it.
I noticed also that user can put custom tags into XML output in gtest right from his tests, gtest provides API for this.
As I mentioned above, I think a better solution is to capture XML output of individual tests and merge them into single report. If test crashed and didn't produce XML report then I can create dummy one. This is not much harder to implement. I'm trying this approach now.

Reply:
Sounds good, I think the less `gtest-parallel` has to invent in general the better (using the existing XMLs sound good to me). Keep --repeat in mind, the existing gtest xml output might provide guidance as it likely handles `--gtest_repeat`.